dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -126,88 +148,115 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
 
     Icon icon = iconService.getIcon(key);
 
-    if (icon instanceof DefaultIcon) {
+    if (!icon.getCustom()) {
       downloadDefaultIcon(icon.getKey(), response);
-    } else if (icon instanceof CustomIcon customIcon) {
-      downloadCustomIcon(customIcon.getFileResourceUid(), response);
+    } else {
+      downloadCustomIcon(icon.getFileResource(), response);
     }
   }
 
-  @GetMapping
-  public @ResponseBody PaginatedIconResponse getAllIcons(IconRequestParams iconRequestParams)
-      throws BadRequestException {
-
-    IconOperationParams iconOperationParams = iconRequestParamsMapper.map(iconRequestParams);
+  @GetMapping(value = "/{key}")
+  public ResponseEntity<Icon> getIconByKey(@PathVariable String key)
+      throws NotFoundException, WebMessageException {
 
-    Pager pager = null;
+    Icon icon = iconService.getIcon(key);
 
-    if (iconRequestParams.isPaging()) {
-      long total = iconService.count(iconOperationParams);
-      pager = new Pager(iconRequestParams.getPage(), total, iconRequestParams.getPageSize());
-      iconOperationParams.setPager(pager);
+    if (icon == null) {
```

Comment:
`getIcon` should throw a `NotFoundException` if the icon is not found. This is directly translated by the `CrudControllerAdvice`. So you don't need to do anything in this case.

Reply:
fixed

Reply:
This is also not fixed. I am still seeing this on the branch when checking out the PR.